src/main/java/org/tron/core/db2/core/SnapshotManager.java

```diff
@@ -198,34 +222,71 @@ public void updateSolidity(int hops) {
   }
 
   private boolean shouldBeRefreshed() {
-    return flushCount >= DEFAULT_FLUSH_COUNT;
+    return flushCount >= maxFlushCount;
   }
 
   private void refresh() {
+    // debug begin
+//    List<List<byte[]>> accounts = new ArrayList<>();
+//    List<String> debugBlockHashs = new ArrayList<>();
+    // debug end
+    List<ListenableFuture<?>> futures = new ArrayList<>(dbs.size());
     for (RevokingDBWithCachingNewValue db : dbs) {
-      if (Snapshot.isRoot(db.getHead())) {
-        return;
-      }
-
-      List<Snapshot> snapshots = new ArrayList<>();
-
-      SnapshotRoot root = (SnapshotRoot) db.getHead().getRoot();
-      Snapshot next = root;
-      for (int i = 0; i < flushCount; ++i) {
-        next = next.getNext();
-        snapshots.add(next);
-      }
+      futures.add(flushServices.get(db.getDbName()).submit(() -> {
+        if (Snapshot.isRoot(db.getHead())) {
+          return;
+        }
 
-      root.merge(snapshots);
+        List<Snapshot> snapshots = new ArrayList<>();
+
+        SnapshotRoot root = (SnapshotRoot) db.getHead().getRoot();
+        Snapshot next = root;
+        for (int i = 0; i < flushCount; ++i) {
+          next = next.getNext();
+          snapshots.add(next);
+          // debug begin
+//        List<byte[]> debugDumpDatas = new ArrayList<>();
+//        String dbName = db.getDbName();
+//        SnapshotImpl snapshot = (SnapshotImpl) next;
+//        DB<Key, Value> keyValueDB = snapshot.getDb();
+//        for (Map.Entry<Key, Value> e : keyValueDB) {
+//          Key k = e.getKey();
+//          Value v = e.getValue();
+//          if ("block".equals(dbName)) {
+//            debugBlockHashs.add(Longs.fromByteArray(k.getBytes()) + ":" + ByteUtil.toHexString(k.getBytes()));
+//          }
+//          if ("account".equals(dbName) && v.getBytes() != null) {
+//            debugDumpDatas.add(v.getBytes());
+//          }
+//        }
+//
+//        if ("account".equals(dbName)) {
+//          accounts.add(debugDumpDatas);
+//        }
+        }
+        root.merge(snapshots);
+        root.resetSolidity();
+        if (db.getHead() == next) {
+          db.setHead(root);
+        } else {
+          next.getNext().setPrevious(root);
+          root.setNext(next.getNext());
+        }
+      }));
+    }
 
-      root.resetSolidity();
-      if (db.getHead() == next) {
-       db.setHead(root);
-      } else {
-        next.getNext().setPrevious(root);
-        root.setNext(next.getNext());
-      }
+    Future<?> future = Futures.allAsList(futures);
+    try {
+      future.get();
+    } catch (InterruptedException | ExecutionException e) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1706&issues=AWb4KHHVsSodgw_KL7wy&open=AWb4KHHVsSodgw_KL7wy)

[ISSUE_KEY]: AWb4KHHVsSodgw_KL7wy